includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1460,20 +1429,112 @@ protected function build_shipping_methods( $shipping_methods ) {
 		return $shipping;
 	}
 
+
+	/**
+	 * Builds the response object to pass to Payment Request
+	 *
+	 * @since 5.3.0
+	 *
+	 * @param bool   $shipping_pending True is the user hasn't defined yet a shipping address.
+	 * @param string $page             Page where the request was made. It can be `product_page` or `cart_page`.
+	 *
+	 * @return array Data object used to update Payment Request button on the client.
+	 */
+	protected function build_response( $shipping_pending = true, $page = 'product_page' ) {
+		if ( 'product_page' == $page ) {
+			$itemized_display_items = true;
+		} elseif ( 'cart_page' == $page ) {
+			$itemized_display_items = false;
+		}
+
+		$data                 = [];
+		$data['currency']     = strtolower( get_woocommerce_currency() );
+		$data['country_code'] = substr( get_option( 'woocommerce_default_country' ), 0, 2 );
+
+		// Taxes
+		$taxes            = WC()->cart->tax_total + WC()->cart->shipping_tax_total;
+		$tax_to_substract = 0;
+
+		if ( wc_tax_enabled() ) {
+			if ( $shipping_pending ) {
+				$tax_to_substract = $taxes;
+				$tax              = 0;
+			}
+		}
+
+		// Shipping
+		$shipping           = wc_format_decimal( WC()->cart->get_shipping_total(), WC()->cart->dp );
+		$shipping_to_remove = 0;
+
+		if ( wc_shipping_enabled() && WC()->cart->needs_shipping() ) {
+			$data['requestShipping'] = ( wc_shipping_enabled() && WC()->cart->needs_shipping() );
+
+			if ( $shipping_pending ) {
+				$shipping_to_remove = $shipping;
+				$shipping           = 0;
+
+				$data['shippingOptions'] = [
+					'id'     => 'pending',
+					'label'  => __( 'Pending', 'woocommerce-gateway-stripe' ),
+					'detail' => '',
+					'amount' => 0,
+				];
+			}
+		}
+
+		// Discounts
+		$discounts = 0;
+
+		if ( version_compare( WC_VERSION, '3.2', '<' ) ) {
+			$discounts = wc_format_decimal( WC()->cart->get_cart_discount_total(), WC()->cart->dp );
+		} else {
+			$applied_coupons = array_values( WC()->cart->get_coupon_discount_totals() );
+
+			foreach ( $applied_coupons as $amount ) {
+				$discounts += (float) $amount;
+			}
+		}
+
+		$discounts = wc_format_decimal( $discounts, WC()->cart->dp );
+
+		// Order total
+		if ( version_compare( WC_VERSION, '3.2', '<' ) ) {
+			$items_total = wc_format_decimal( WC()->cart->cart_contents_total, WC()->cart->dp ) + $discounts;
+			$order_total = wc_format_decimal( $items_total + $tax + $shipping - $discounts, WC()->cart->dp );
+		} else {
+			// Getting the total amount from the cart automatically adds a shipping cost to it
+			// We need to remove it if the user hasn't picked a shipping addres yet
+			$order_total = WC()->cart->get_total( false ) - $tax_to_substract - $shipping_to_remove;
```

Comment:
This doesn't look too good but I haven't found a way to to get the cart total without tax and shipping.